### PR TITLE
move activation dispatches into helper function

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/pull/3302

### CODE REVIEW

File: src/diffusers/models/resnet.py [hunk 1]
```diff
@@ -646,11 +640,6 @@ def forward(self, input_tensor, temb):
         return output_tensor
 
 
-class Mish(torch.nn.Module):
-    def forward(self, hidden_states):
-        return hidden_states * torch.tanh(torch.nn.functional.softplus(hidden_states))
-
```
Comment (williamberman) [inline src/diffusers/models/resnet.py:652]: Not used

File: src/diffusers/models/activations.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from torch import nn
+
+
+def get_activation(act_fn):
+    if act_fn in ["swish", "silu"]:
```
Comment (hypnopump) [inline src/diffusers/models/activations.py:5]:
```suggestion
    if act_fn in {"swish", "silu"}:
```

Comment (hypnopump) [reply src/diffusers/models/activations.py:5]: no need for order here

**Review by hypnopump — APPROVED** [review]
lgtm

**Review by sayakpaul — APPROVED** [review]
Short and clean refactor! Thanks.

**Review by pcuenca — APPROVED** [review]

**Review by patrickvonplaten — APPROVED** [review]
Great clean-up! This will help a lot :-) 

Could you add some quick tests as well?

**Comment by williamberman** [comment]
> Great clean-up! This will help a lot :-)
> 
> Could you add some quick tests as well?

done!

**Pull request merged**